### PR TITLE
Weakens mining armour while in high pressure environments

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Mining armour is ridiculously strong to deal with the high damages of lavaland, however, this creates a huge imbalance when used on the station where everything doesn't deal very high damage. This means miners are very strong and are very difficult to stop, leading to them spending a large portion of the round on station hunting down antags that don't have much of a chance fighting them.

This PR adds in a way to modify the strength of armour under certain conditions as well as making it so mining armour is weakened in high-pressure environments.

## Why It's Good For The Game

Weakens the extremely strong mining armour without making them completely worthless on lavaland.

## Changelog
:cl:
balance: Mining armour is now weaker in high pressure environments.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
You forgot explorer suit, once it's fully upgraded with goliath plates it's only 10 melee armor less than ash drake armor. Not sure how the armor upgrade interacts with this mechanic, it also applies to mining hardsuit.

Not that it would make sense for the goliath plates to be weaker, but hey none of these changes make sense and are just for balance.

**Comment by Sapfearon** [comment]
"miners are very strong and are very difficult to stop" - bacon please. Stunbaton stops any miner, disabler works even better. Also tasers/hos gun. I mean, basic suit/hardsuit is a bit better then armor vest for brute but has worse burn resist. You can upgrade it 3 times for extra 30% which makes it 2nd or 3rd best armor against brute. But alright, you can nerf basic suit for sure. However hardsuits - all of them are fine. Well you can argue that it's too east to get mining hardsuit which is true, but nobody ever upgrades it so who cares? Even engi hardsuit has decent stats if you never noticed.

And for love of god do not nerf ashdrake armor or heck suit. Ashdrake armor was nerfed already and heck is hardest to get suit in the game ( which also worse then most of suits including drake armor, lol).  Not to mention slugs still 2-3 shot crit you. Talking about slugs - your shotgun nerf broke combat shotguns. All of them, even warden combat shotty. They now need to be pumped after every shot which is stupid. They literally can't be pumped, they are not pump action.

**Comment by WhyIsCaeciliusTaken** [comment]
> "miners are very strong and are very difficult to stop" - bacon please. Stunbaton stops any miner, disabler works even better. Also tasers/hos gun. I mean, basic suit/hardsuit is a bit better then armor vest for brute but has worse burn resist. You can upgrade it 3 times for extra 30% which makes it 2nd or 3rd best armor against brute. But alright, you can nerf basic suit for sure. However hardsuits - all of them are fine. Well you can argue that it's too east to get mining hardsuit which is true, but nobody ever upgrades it so who cares? Even engi hardsuit has decent stats if you never noticed.
> 
> And for love of god do not nerf ashdrake armor or heck suit. Ashdrake armor was nerfed already and heck is hardest to get suit in the game ( which also worse then most of suits including drake armor, lol). Not to mention slugs still 2-3 shot crit you. Talking about slugs - your shotgun nerf broke combat shotguns. All of them, even warden combat shotty. They now need to be pumped after every shot which is stupid. They literally can't be pumped, they are not pump action.

noooo I can't validhunt on station with my overpowered mining gear as efficiently anymore nooooo!

**Comment by PowerfulBacon** [comment]
> 
> 
> "miners are very strong and are very difficult to stop" - bacon please. Stunbaton stops any miner, disabler works even better. Also tasers/hos gun. I mean, basic suit/hardsuit is a bit better then armor vest for brute but has worse burn resist. You can upgrade it 3 times for extra 30% which makes it 2nd or 3rd best armor against brute. But alright, you can nerf basic suit for sure. However hardsuits - all of them are fine. Well you can argue that it's too east to get mining hardsuit which is true, but nobody ever upgrades it so who cares? Even engi hardsuit has decent stats if you never noticed.
> 
> And for love of god do not nerf ashdrake armor or heck suit. Ashdrake armor was nerfed already and heck is hardest to get suit in the game ( which also worse then most of suits including drake armor, lol). Not to mention slugs still 2-3 shot crit you. Talking about slugs - your shotgun nerf broke combat shotguns. All of them, even warden combat shotty. They now need to be pumped after every shot which is stupid. They literally can't be pumped, they are not pump action.

The HECK suit and ash drake armour have less decreases in pressure resistance to account for them being harder to get, however in their current state 70 melee resistance is insane, considering that deathsquads (made to be nearly invincible) have 80. The values are high because lavaland damages are very high, however on station the damages are significantly lower because killing people in 3 shots with a toolbox would be slightly less fun. This PR accounts for the imbalance between lavaland and station damages by lowering the resistance of the armour in the place where the damages are lower.
Note that all the hardsuits will still be as strong in space as on lavaland, so they aren't completely useless on station. This is great since it means using the armour defensively can empower you a lot, making them good for antags, while using them aggressively and pushing into areas of high pressure makes them a lot weaker, nerfing them when used for validhunting.

**Comment by PowerfulBacon** [comment]
Also at the moment stun batons follow melee resistance, meaning the ash drake armour will drop the stam damage by 70% too

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -41,6 +41,8 @@
 	var/dynamic_hair_suffix = ""//head > mask for head hair
 	var/dynamic_fhair_suffix = ""//mask > head for facial hair
 
+	var/high_pressure_multiplier = 1
+	var/list/high_pressure_multiplier_types = list("melee", "bullet", "laser", "energy", "bomb")
```
Comment (park66665) [inline code/modules/clothing/clothing.dm]: Maybe this can be a static list?

File: code/modules/antagonists/cult/cult_items.dm [hunk 2]
```diff
@@ -327,18 +327,19 @@
...
 	brightness_on = 0
 	actions_types = list()
+	high_pressure_multiplier = 0.5
 
 /obj/item/clothing/suit/space/hardsuit/cult
 	name = "\improper Nar'Sien hardened armor"
 	icon_state = "cult_armor"
 	item_state = "cult_armor"
-	desc = "A heavily-armored exosuit worn by warriors of the Nar'Sien cult. It can withstand hard vacuum."
+	desc = "A heavily-armored exosuit worn by warriors of the Nar'Sien cult. It is reinforced by hard vacuum."
```
Comment (park66665) [inline code/modules/antagonists/cult/cult_items.dm]: Apparently you forgot to set `high_pressure_multiplier` for this... or you didn't?

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -331,3 +333,15 @@ BLIND     // can't see anything
 		deconstruct(FALSE)
 	else
 		..()
+
+/obj/item/clothing/get_armor_rating(d_type, mob/wearer)
+	. = ..(d_type, wearer)
```
Comment (park66665) [inline code/modules/clothing/clothing.dm]:
```suggestion
	. = ..()
```

I don't think explicit argument passing is needed.

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -331,3 +333,15 @@ BLIND     // can't see anything
...
+	if(high_pressure_multiplier == 1)
+		return
+	var/turf/T = get_turf(wearer)
+	if(!T)
+		return
+	if(!(d_type in high_pressure_multiplier_types))
+		return
```
Comment (park66665) [inline code/modules/clothing/clothing.dm]:
```suggestion
	if(!T || !(d_type in high_pressure_multiplier_types))
		return
```

**Comment by Sapfearon** [comment]
> noooo I can't validhunt on station with my overpowered mining gear as efficiently anymore nooooo!

Sec disabler/stunbaton/taser/AP ammo can deal with any armor without problem. Traitors have even better time due to stronger weapons and contractor batons/alien batons.
So sorry but, against what this nerf exactly? If this is nerf to traitor miner then it's pointless.  
Most of the time when you valid as miner it's red alert and station probably has 0 atmos which means your nerf is pointless bacon.  I fail to see why you need to nerf mining suits instead of removing completly broken and OP items from tendril chests like grenade, katana, hook and such. I'd even argue that spellblade indeed needs nerfs, maybe lower delimb chance or something but nerfing heck suit, which is weakest part of loot does not make sense for me. In fact, you never kill ashdrake/bubble for suit, you want that sweet drake form/spellblade/staff or something. Suit is just nice bonus.

**Comment by PowerfulBacon** [comment]
> 
> 
> > noooo I can't validhunt on station with my overpowered mining gear as efficiently anymore nooooo!
> 
> Sec disabler/stunbaton/taser/AP ammo can deal with any armor without problem. Traitors have even better time due to stronger weapons and contractor batons/alien batons.
> So sorry but, against what this nerf exactly? If this is nerf to traitor miner then it's pointless.
> Most of the time when you valid as miner it's red alert and station probably has 0 atmos which means your nerf is pointless bacon. I fail to see why you need to nerf mining suits instead of removing completly broken and OP items from tendril chests like grenade, katana, hook and such. I'd even argue that spellblade indeed needs nerfs, maybe lower delimb chance or something but nerfing heck suit, which is weakest part of loot does not make sense for me. In fact, you never kill ashdrake/bubble for suit, you want that sweet drake form/spellblade/staff or something. Suit is just nice bonus.

Then I shall nerf them too. Other items being better isn't an excuse for everything else to be unbalanced.

**Comment by Sinestia** [comment]
A space nerf seems very necessary too.


 I'd say actually the most common validhunting miners do is in space, not on station.

To be fair this was prompted by observing one singlehandedly kill all 3 pirates, their turrets, while taking little damage from direct hits and lasers. (Granted she also had a katana, which is 10 damage more than an esword.) An "i ded fuck this" by proxy.

File: code/modules/antagonists/cult/cult_items.dm (hunk 2 above)
Comment (PowerfulBacon) [reply code/modules/antagonists/cult/cult_items.dm]: Oh, I updated the description of the parent type by accident oops.

File: code/modules/clothing/clothing.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/clothing/clothing.dm]: I'd say thats a good idea, although then the list likely wouldn't be changeable for other types (Although those other types don't make sense anyway)

**Comment by PowerfulBacon** [comment]
Will make the requested changes at somepoint and get the mining hardsuit too. Nerfing them in space is somewhat unviable since the lavaland pressure check is for pressure and wouldn't make sense being metalocked to the z-level.

**Comment by Sinestia** [comment]
If only lavaland was actually really hot, because that'd be the most sensible way

oh well, it's still a start

**Comment by PowerfulBacon** [comment]
Ill do the mining hardsuit or whatenver needed doing now

**Comment by Rukofamicom** [comment]
wouldn't it be better to reduce the power of the armor everywhere and also balance the lavaland mobs so we don't need awkward scaling?

**Comment by Isy232** [comment]
I think locking their stats to the Z level is fine.

Make up some "they take power from the necropolis" or something.

I feel miners will just make holes in the station so their mining gear is still OP.

**Comment by Sapfearon** [comment]
Again i fail to understand need to nerf armor. If miner validhunts then a) it's against rules b) he is traitor c) given that's mining is only job who actually has weapons and guns it makes sense to try and save station 
So as people already pointed out, PKA does double damage in low pressure, up to 40-60 damage per shot ( depends on mods.) Any decent miner always dual wield them, dealing 80+ damage per hit if some idiot decided to fight them in low pressure. But honestly, aside from space pirates, space dragon, blob and xenos i don't think you will see anyone using pka. Crusher already does not work on like 90% living beings even if it clearly SHOULD. And pka dealing insane damage in low pressure is feature, not a bug.
Armor is not problem, real problem is absolutely moronic loot in tendrils which is pretty much for traitors only. I never ever pick up crap like hook, grenade katanas or such, since i don't need it for my job. In fact, aside from suits/holopara/eye of god everything else is either pure crap or OP item for traitor for free. So by nerfing armor you just...nerfing only decent loot you can get from tendrils/bosses with guarantee? 
I'm also curious when you even saw validhunting miner? when shift lasted longer then 2h? Do we need to nerf something for 2h+ shifts? You spend 15-20 minutes to get advanced mining/bag, then about 10 min to clear first boss, couple tendrils, get bag of holding. Then you start killing bosses like drakes, bubble, colossus and such. That takes time and preparation and when you finally killed strongest mobs in game all you get is 3 disabler shots and instant death most of time. Not even holopara can protect you against stamina damage. Remember that legion cores were already nerfed 2 times. They are no longer aheal, and they work at 50% outside of lavaland. So even if you have best armor you can quickly die to...heretic with his stuns, lings with his stungs, contractor, abductor, anyone with stunbaton/disabler/flashbang. And that's assuming we talking about flashproof suits which drake armor/normal suit/goliath cape are not. Oh and piercing syringes, explosives, magic still work. Plus you can just od miner with 2 medipens which insta kills miner without any hope for revival without loong surgery to filter blood and replace liver.

Easy solution to fix validhunting idiots in good armor is to remove that armor from chests completly and give it to bosses. Inquisitor suit can be colossus loot, blood drunk can have that red hardsuit. So now if you for some ungodly reason want to validhunt ( which is against server rules already, ahem) you need to put some effort into it. 
I just feel if you spend 1-2 hours clearing whole lavaland and killing all bosses you deserve swag gear and armor is least harmful gear you can swag with.  So before you jump into nerfing more shit, do some balancing first. I absolutly hate your shotgun nerf and i dreading to see any more nerfs from you ( you can't even pump shotgun while holding slug in other hand to load it in. I can do it irl but in game? nope, too overpowered i guess.)  Right now as sec you just order all autorifles in the world and hope they still not nerfed.

**Comment by PowerfulBacon** [comment]
> Easy solution to fix validhunting idiots in good armor is to remove that armor from chests completly and give it to bosses. 

Thats where most of the armour comes from (Ash drake which is one of the best for it at least).

>  So before you jump into nerfing more shit, do some balancing first. 

Thats... literally what nerfing is

> ( you can't even pump shotgun while holding slug in other hand to load it in. I can do it irl but in game? nope, too overpowered i guess.)

I guess you have a point here
https://www.youtube.com/watch?v=mOf4-2Ixhbg

> You spend 15-20 minutes to get advanced mining/bag, then about 10 min to clear first boss

Hierophant is easily doable in the first 5 minutes with roundstart gear, ash drake isn't too difficult either for most robust miners.

> nerfing only decent loot you can get from tendrils/bosses with guarantee?

AKA nerfing the best loot, since all the others have arguably better functions however their function is situational.

> And pka dealing insane damage in low pressure is feature, not a bug.

I never implied it was a bug in any way, in fact this builds on the mechanic that miner gear is better in low pressures by making armour have insane armour values in low pressures.

> So even if you have best armor you can quickly die to...heretic with his stuns, lings with his stungs, contractor, abductor, anyone with stunbaton/disabler/flashbang.

People die to antags in deathsquad gear, that doesn't mean deathsquad gear is weak.

**Comment by LivingBatteryBox** [comment]
Honestly we just need to overhaul lavaland balance in general. We just need a rework like the keys and tokens reworks that were/are being worked on.

**Comment by Sapfearon** [comment]
You forgot something bacon. Ashdrake armor is no longer hardsuit nor it's pressure proof. You already can take damage from pressure in lavaland which sometimes happends and annoying as f.  Plus, yes it's easy for me to bum rush hiero with 1 medipen or kill drake in melee but... As turns out, about 80% of miners can't kill even 1 boss. Yes, i had round where i killed 5 ashdrakes but who cares if most miners can't kill single drake? Someone killing bubble is rare as hell, even i rarely kill him with my 200h on miner. In fact i know only 2 player who could kill bubble every shift and they both banned from sage. So nerfing item which only at best 20% of people can use, then given how rare for me to see valihunting miner it's what, maybe 10% of those people who can get that armor?

Armor does not enables or helps with validhunting, OP weapons that you can get for absolutly no effort from tendrils are real problem. I suggest start by removing most items from tendrils and replacing them with loot which you actually want to use for mining. Like i would trade katana or hook for a simple bag of holding any day. 

Also how often you saw me validhunting when i'm miner? i'm really curious. I'm Max Anders in game.

P.s. isn't sarah connor in t2 pumping shotgun with 1 hand in ending?...

**Comment by PowerfulBacon** [comment]
:trollface:

**Review by utilbird — APPROVED** [review]

**Pull request merged**